starboard/rdk/launcher.py

```diff
@@ -0,0 +1,426 @@
+  _PEXPECT_READLINE_TIMEOUT_MAX_RETRIES = 900
+  # Delay between subsequent SSH commands
+  _INTER_COMMAND_DELAY_SECONDS = 1.5
+
+  # This is used to strip ansi color codes from pexpect output.
+  _PEXPECT_SANITIZE_LINE_RE = re.compile(r'\x1b[^m]*m')
+
+  # Exceptions to retry
+  _RETRY_EXCEPTIONS = (pexpect.TIMEOUT, pexpect.ExceptionPexpect,
+                       pexpect.exceptions.EOF, OSError)
+
+  def __init__(self, platform, target_name, config, device_id, **kwargs):
+    # pylint: disable=super-with-arguments
+    super().__init__(platform, target_name, config, device_id, **kwargs)
+    env = os.environ.copy()
+    env.update(self.env_variables)
+    self.full_env = env
+    self.platform = platform
+
+    if not self.device_id:
+      self.device_id = self.full_env.get('RDK_ADDR')
+      if not self.device_id:
+        raise ValueError(
+            'Unable to determine target, please pass it in, or set RDK_ADDR '
+            'environment variable.')
+
+    self.startup_timeout_seconds = Launcher._STARTUP_TIMEOUT_SECONDS
+
+    self.pexpect_process = None
+
+    self._InitPexpectCommands()
+
+    self.run_inactive = threading.Event()
+    self.run_inactive.set()
+
+    self.shutdown_initiated = threading.Event()
+
+    self.log_targets = kwargs.get('log_targets', True)
+
+    signal.signal(signal.SIGINT, functools.partial(_sigint_or_sigterm_handler))
+    signal.signal(signal.SIGTERM, functools.partial(_sigint_or_sigterm_handler))
+
+    self.last_run_pexpect_cmd = ''
+
+  def _InitPexpectCommands(self):
+    """Initializes all of the pexpect commands needed for running the test."""
+
+    # Ensure no trailing slashes
+    self.out_directory = self.out_directory.rstrip('/')
+
+    rdk_user_hostname = Launcher._RDK_USERNAME + '@' + self.device_id
+    rdk_test_dir = '/usr/share/content/data/app'
+
+    # scp command setup
+    options = '-rOCq '
+    rsa_options = '-o \"LogLevel ERROR\" \
+        -o \"UserKnownHostsFile=/dev/null\" -o StrictHostKeyChecking=no '
+
+    source = os.path.join(self.out_directory, 'content', 'app', 'cobalt')
+    destination = f'{rdk_user_hostname}:{rdk_test_dir}/'
+    self.scp_command = 'scp ' + options + rsa_options + \
+        ' ' + source + ' ' + destination
+
+    # ssh command setup
+    self.ssh_command = 'ssh -t ' + rsa_options + rdk_user_hostname + \
+        ' TERM=dumb bash -l'
+
+    # test output tags
+    self.test_complete_tag = 'test suites ran.'
+    self.test_failure_tag = 'tests, listed below'
+    self.test_failure_list_tag = 'FAILED TESTS'
+
+    # test command setup
+    # note: If RDK has ResidentApp feature turned on (default), the log
+    #       goes to a buffer that can be retrieved by command
+    #           journalctl -f -u dobby
+    #       Otherwise, the container is off, and the log is dumped into
+    #       file /opt/logs/wpeframework.log.
+    #           tail -f /opt/logs/wpeframework.log
+    self.test_command = 'curl -X POST http://127.0.0.1:9998/jsonrpc -d \
```

Comment:
Can you break out the values in this string to variables or constants (as applicable)?

Reply:
I'm not asking you to change the content of the string, just make it more readable. The line currently has two separate commands that are lumped into one, the curl command (which I'm assuming is triggering the loading of `libcobalt.so` and the tail command to view the logs. It also contains some hardcoded (I assume?) json fields. 

These could be broken up into variables and put back together in a format string.

Reply:
Correct. I am working on that.

Reply:
Sorry that I didn't catch your idea yesterday. Here it is. Also once we sort out the argument issue with RDK, more variables will be added into the command line in this fashion.